File: 12_06_2024_Sort_Colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        mp = {}
        for i in nums:
            if i in mp:
                mp[i] += 1
            else:
                mp[i] = 1
        
        idx = 0
        for key in range(3):  # We know the keys are 0, 1, 2
            count = mp.get(key, 0)  # Use get to handle missing keys
            for _ in range(count):
                nums[idx] = key
                idx += 1

File: 12_06_2024_Sort_Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (int i : nums) {
            mp[i]++;
        }
        
        int idx = 0;
        for (int key = 0; key < 3; key++) {  // We know the keys are 0, 1, 2
            int count = mp.count(key) ? mp[key] : 0;  // Use count to handle missing keys
            for (int _ = 0; _ < count; _++) {
                nums[idx++] = key;
            }
        }
    }
};

